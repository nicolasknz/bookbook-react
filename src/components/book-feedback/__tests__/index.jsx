import React from 'react';
import renderer from 'react-test-renderer';
import BookFeedback from '../index';
import * as redux from 'react-redux'

const spy = jest.spyOn(redux, 'useSelector')
spy.mockReturnValue({ session: 'test' })

describe("Render test", () => {
  it('renders correctly', () => {
    const tree = renderer

      .create(<BookFeedback book={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const book = { id: 1, title: "test", img_url: "img" }
    const tree = renderer

      .create(<BookFeedback book={book} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

// describe("test callBack", () => {
//   it('renders correctly callBack', () => {
//     const mockCallback = jest.fn()
//     const wrapper = shallow(<BookFeedback book={book} onClick={mockCallback} />)

//     wrapper.find("ModalButton").at(1).simulate("click");

//     expect(mockCallback).toHaveBeenCalledWith(2);
//     expect(mockCallback).toHaveBeenCalled();

//   });
// });