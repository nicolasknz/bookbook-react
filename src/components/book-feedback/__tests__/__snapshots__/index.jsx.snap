// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render test renders correctly 1`] = `
<button
  className="sc-bdnylx sc-gtssRu jayiDO ZgWam"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Avaliar
</button>
`;

exports[`Render test renders correctly 2`] = `
<button
  className="sc-bdnylx sc-gtssRu jayiDO ZgWam"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Avaliar
</button>
`;
